src/transform.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <glm/glm.hpp>
+
+/*
+    Handles all matrix transformations for a game object. Gives a (hopefully) simple
+    interface for interacting with the game object's transformation, including
+    translations, rotations, and scaling. 
+*/
+class Transform {
+public:
+    Transform();    
+
+    void set_world_mat(glm::mat4 mat);
+
+    void set_position(glm::vec3 position);
```

Comment:
May be better to enforce a single call signature. We can emulate the 3 float style like so, thanks to implicit constructors:

```c++
set_position({x, y, z});
```